arangod/Scheduler/SchedulerFeature.cpp

```diff
@@ -47,24 +49,41 @@ using namespace arangodb::basics;
 using namespace arangodb::options;
 using namespace arangodb::rest;
 
+namespace {
+  /// @brief return the default number of threads to use (upper bound)
+  size_t defaultNumberOfThreads() {
+    // use two times the number of hardware threads as the default
+    size_t result = TRI_numberProcessors() * 2;
+    // but only if higher than 64. otherwise use a default minimum value of 64
+    if (result < 64) {
+      result = 64;
```

Comment:
IMHO 64 is way to high for a conventional machine (e.g. 4 cores).

Reply:
Not with out blocking threads arch in arangodb. This is required until things like ClusterComm are completely async.

Reply:
I know that threads can block in the current architecture, and you need additional threads to compensate this. You obviously have more insight than me, but 64 still feels pretty high. I would probably cap the number of threads by 8 per core.